tracing-api/src/main/java/com/palantir/tracing/api/TracingHeadersEnrichingFunction.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tracing.api;
+
+/** Function used to adapt the tracing library to any arbitrary request type. */
+public interface TracingHeadersEnrichingFunction<T> {
```

Comment:
This class should live in the `tracing` module (implementation), not tracing-api.